### DISCUSSION TITLE
[crazyswarm_server-1] process has died

### DISCUSSION THREAD

**Discussion by nfu64967512** [USC-ACTLab/crazyswarm#782] [Q&A] [answered]
Hello,

I've recently embarked on a project involving a Crazyswarm drone performance. However, I've encountered some issues when running the command roslaunch /home/bitcraze/crazyswarm/ros_ws/src/crazyswarm/launch/hover_swarm.launch. Below is the output I received:
bitcraze@ubuntu:~/crazyswarm$ roslaunch /home/bitcraze/crazyswarm/ros_ws/src/crazyswarm/launch/hover_swarm.launch
... logging to /home/bitcraze/.ros/log/b82dc00c-a495-11ee-b779-13d910275467/roslaunch-ubuntu-7881.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://ubuntu:39587/

SUMMARY
========

PARAMETERS
 * /crazyflieTypes/CF21SingleMarker/batteryVoltageWarning: 3.8
 * /crazyflieTypes/CF21SingleMarker/batteryVoltateCritical: 3.7
 * /crazyflieTypes/CF21SingleMarker/bigQuad: False
 * /crazyflieTypes/CF21SingleMarker/dynamicsConfiguration: 0
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/i_range_z: 0.4
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kR_xy: 70000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_xy: 0.2
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kd_z: 0.4
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_xy: 0.05
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/ki_z: 0.05
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kp_xy: 0.4
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kp_z: 1.25
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kw_xy: 20000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/mass: 0.032
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/ctrlMel/massThrust: 132000
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/mNGyro_rollpitch: 0.1
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/mNGyro_yaw: 0.1
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/pNAcc_xy: 1.0
 * /crazyflieTypes/CF21SingleMarker/firmwareParams/kalman/pNAcc_z: 2.0
 * /crazyflieTypes/CF21SingleMarker/markerConfiguration: 3
 * /crazyflieTypes/default/batteryVoltageWarning: 3.8
 * /crazyflieTypes/default/batteryVoltateCritical: 3.7
 * /crazyflieTypes/default/bigQuad: False
 * /crazyflieTypes/default/dynamicsConfiguration: 0
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/default/firmwareParams/ctrlMel/i_range_z: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kR_xy: 70000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_xy: 0.2
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kd_z: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_xy: 0.05
 * /crazyflieTypes/default/firmwareParams/ctrlMel/ki_z: 0.05
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kp_xy: 0.4
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kp_z: 1.25
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kw_xy: 20000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/default/firmwareParams/ctrlMel/mass: 0.032
 * /crazyflieTypes/default/firmwareParams/ctrlMel/massThrust: 132000
 * /crazyflieTypes/default/markerConfiguration: 0
 * /crazyflieTypes/defaultSingleMarker/batteryVoltageWarning: 3.8
 * /crazyflieTypes/defaultSingleMarker/batteryVoltateCritical: 3.7
 * /crazyflieTypes/defaultSingleMarker/bigQuad: False
 * /crazyflieTypes/defaultSingleMarker/dynamicsConfiguration: 0
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/i_range_z: 0.4
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kR_xy: 70000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_xy: 0.2
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kd_z: 0.4
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_xy: 0.05
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/ki_z: 0.05
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kp_xy: 0.4
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kp_z: 1.25
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kw_xy: 20000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/mass: 0.032
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/ctrlMel/massThrust: 132000
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/mNGyro_rollpitch: 0.2
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/mNGyro_yaw: 0.1
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/pNAcc_xy: 4.0
 * /crazyflieTypes/defaultSingleMarker/firmwareParams/kalman/pNAcc_z: 8.0
 * /crazyflieTypes/defaultSingleMarker/markerConfiguration: 3
 * /crazyflieTypes/large/batteryVoltageWarning: 11.4
 * /crazyflieTypes/large/batteryVoltateCritical: 11.1
 * /crazyflieTypes/large/bigQuad: True
 * /crazyflieTypes/large/dynamicsConfiguration: 0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/i_range_z: 4.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kR_xy: 30000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kR_z: 40000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_omega_rp: 100
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_xy: 3.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kd_z: 3.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_xy: 1.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/ki_z: 2.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kp_xy: 8.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kp_z: 8.0
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kw_xy: 9000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/kw_z: 10000
 * /crazyflieTypes/large/firmwareParams/ctrlMel/mass: 0.38
 * /crazyflieTypes/large/firmwareParams/ctrlMel/massThrust: 5400
 * /crazyflieTypes/large/markerConfiguration: 2
 * /crazyflieTypes/medium/batteryVoltageWarning: 7.6
 * /crazyflieTypes/medium/batteryVoltateCritical: 7.4
 * /crazyflieTypes/medium/bigQuad: True
 * /crazyflieTypes/medium/dynamicsConfiguration: 0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_m_z: 1500
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_xy: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/i_range_z: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kR_xy: 50000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kR_z: 60000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_omega_rp: 200
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_xy: 1.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kd_z: 1.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_m_z: 500
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_xy: 0.1
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/ki_z: 0.5
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kp_xy: 2.0
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kp_z: 3
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kw_xy: 16000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/kw_z: 12000
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/mass: 0.11
 * /crazyflieTypes/medium/firmwareParams/ctrlMel/massThrust: 23000
 * /crazyflieTypes/medium/markerConfiguration: 1
 * /crazyflies: [{'channel': 80, ...
 * /crazyswarm_server/enable_logging: True
 * /crazyswarm_server/enable_logging_pose: True
 * /crazyswarm_server/enable_parameters: True
 * /crazyswarm_server/firmwareParams/commander/enHighLevel: 1
 * /crazyswarm_server/firmwareParams/kalman/resetEstimation: 1
 * /crazyswarm_server/firmwareParams/locSrv/extPosStdDev: 1e-3
 * /crazyswarm_server/firmwareParams/locSrv/extQuatStdDev: 0.05
 * /crazyswarm_server/firmwareParams/ring/effect: 16
 * /crazyswarm_server/firmwareParams/ring/headlightEnable: 0
 * /crazyswarm_server/firmwareParams/ring/solidBlue: 255
 * /crazyswarm_server/firmwareParams/ring/solidGreen: 0
 * /crazyswarm_server/firmwareParams/ring/solidRed: 0
 * /crazyswarm_server/firmwareParams/stabilizer/controller: 2
 * /crazyswarm_server/firmwareParams/stabilizer/estimator: 2
 * /crazyswarm_server/force_no_cache: False
 * /crazyswarm_server/genericLogTopicFrequencies: [10]
 * /crazyswarm_server/genericLogTopic_log1_Variables: ['stateEstimate.x...
 * /crazyswarm_server/genericLogTopics: ['log1']
 * /crazyswarm_server/motion_capture_host_name: vicon
 * /crazyswarm_server/motion_capture_type: none
 * /crazyswarm_server/object_tracking_type: libobjecttracker
 * /crazyswarm_server/print_latency: False
 * /crazyswarm_server/save_point_clouds: /dev/null
 * /crazyswarm_server/send_position_only: False
 * /crazyswarm_server/write_csvs: False
 * /crazyswarm_teleop/csv_file: /home/bitcraze/cr...
 * /crazyswarm_teleop/timescale: 0.8
 * /dynamicsConfigurations/0/maxFitnessScore: 0.001
 * /dynamicsConfigurations/0/maxPitch: 1.4
 * /dynamicsConfigurations/0/maxPitchRate: 20.0
 * /dynamicsConfigurations/0/maxRoll: 1.4
 * /dynamicsConfigurations/0/maxRollRate: 20.0
 * /dynamicsConfigurations/0/maxXVelocity: 2.0
 * /dynamicsConfigurations/0/maxYVelocity: 2.0
 * /dynamicsConfigurations/0/maxYawRate: 10.0
 * /dynamicsConfigurations/0/maxZVelocity: 3.0
 * /joy/dev: /dev/input/js0
 * /markerConfigurations/0/numPoints: 4
 * /markerConfigurations/0/offset: [0.0, -0.01, -0.04]
 * /markerConfigurations/0/points/0: [0.0177184, 0.013...
 * /markerConfigurations/0/points/1: [-0.0262914, 0.05...
 * /markerConfigurations/0/points/2: [-0.0328889, -0.0...
 * /markerConfigurations/0/points/3: [0.0431307, -0.03...
 * /markerConfigurations/1/numPoints: 4
 * /markerConfigurations/1/offset: [0.0, 0.0, -0.03]
 * /markerConfigurations/1/points/0: [-0.00896228, -0....
 * /markerConfigurations/1/points/1: [-0.0156318, 0.09...
 * /markerConfigurations/1/points/2: [0.0461693, -0.08...
 * /markerConfigurations/1/points/3: [-0.0789959, -0.0...
 * /markerConfigurations/2/numPoints: 4
 * /markerConfigurations/2/offset: [0.0, 0.0, -0.06]
 * /markerConfigurations/2/points/0: [0.0558163, -0.00...
 * /markerConfigurations/2/points/1: [-0.0113941, 0.00...
 * /markerConfigurations/2/points/2: [-0.0306277, 0.05...
 * /markerConfigurations/2/points/3: [0.0535816, -0.04...
 * /markerConfigurations/3/numPoints: 1
 * /markerConfigurations/3/offset: [0.0, -0.01, -0.04]
 * /markerConfigurations/3/points/0: [0.0177184, 0.013...
 * /numDynamicsConfigurations: 1
 * /numMarkerConfigurations: 4
 * /rosdistro: noetic
 * /rosversion: 1.16.0

NODES
  /
    crazyswarm_server (crazyswarm/crazyswarm_server)
    crazyswarm_teleop (crazyswarm/crazyswarm_teleop)
    joy (joy/joy_node)
    rviz (rviz/rviz)

ROS_MASTER_URI=http://localhost:11311

process[crazyswarm_server-1]: started with pid [7901]
process[joy-2]: started with pid [7902]
process[crazyswarm_teleop-3]: started with pid [7903]
[ WARN] [1703676633.313436592]: Couldn't set gain on joystick force feedback: Bad file descriptor
[ INFO] [1703676633.317425727]: Opened joystick: /dev/input/js0 (VirtualBox USB Tablet). deadzone_: 0.050000.
process[rviz-4]: started with pid [7909]
[ INFO] [1703676633.430792609]: Wait for services...
[ INFO] [1703676633.436511560]: Manager ready.
ch: 1
[ INFO] [1703676633.536710001]: Parsed crazyflies.yaml successfully.
[ INFO] [1703676633.538220056]: Adding CF: cf1 (radio://0/80/2M/E7E7E7E701, cf1)...
[ INFO] [1703676633.578085501]: CF ctor: 0.039741 s
[ INFO] [1703676633.605823866]: [cf1] ---
[ INFO] [1703676633.624342659]: [cf1] SYS: Crazyflie 2.1 is up and running!
[ INFO] [1703676633.646609753]: [cf1] SYS: Build 0:f83e4269f59b (2023.07) CLEAN
[ INFO] [1703676633.670254252]: [cf1] SYS: I am 0x313232313232470F0035002F and I have 1024KB of flash!
[ INFO] [1703676633.680650151]: [cf1] CFGBLK: v1, verification [OK]
[ INFO] [1703676633.690289479]: [cf1] DECK_CORE: 2 deck(s) found
[ INFO] [1703676633.741098644]: [cf1] DECK_CORE: Calling INIT on driDECK_CORE: Calling INIT on driIMU: BMI088: Using I2C interface.
[ INFO] [1703676633.759705122]: [cf1] K].
[ INFO] [1703676633.779769526]: [cf1] IMU: BMI088 Accel connection [OK]
[ INFO] [1703676633.797418795]: [cf1] IMU: BMP388 I2C connection [OK]
[ INFO] [1703676633.815919135]: [cf1] ESTIMATOR: Using Kalman (2) estimator
[ INFO] [1703676633.834418387]: [cf1] CONTROLLER: Using PID (1) controller
[ INFO] [1703676633.882966709]: [cf1] MTR-DRV: Using brushed motor de. NRF51 TX power offset by -12db.
[ INFO] [1703676633.903338675]: [cf1] SYS: About to run tests in system.c.
[ INFO] [1703676633.913227807]: [cf1] SYS: NRF51 version: 2023.02
[ INFO] [1703676633.922905093]: [cf1] EEPROM: I2C connection [OK].
[ INFO] [1703676633.932662943]: [cf1] STORAGE: Storage check [OK].
[ INFO] [1703676633.951261196]: [cf1] IMU: BMI088 gyro self-test [OK]
[ INFO] [1703676633.962870522]: [cf1] DECK_CORE: Deck 0 test [OK].
[ INFO] [1703676633.971755026]: [cf1] SYS: Self test passed!
[ INFO] [1703676633.987971518]: [cf1] STAB: Wait for sensor calibration...
[ INFO] [1703676633.998759083]: [cf1] SYS: Free heap: 17680 bytes
[ INFO] [1703676634.017263140]: [cf1] STAB: Starting stabilizer loop
[ INFO] [1703676634.036885714]: [cf1] ESTKALMAN: State out of bounds, resetting
[ INFO] [1703676634.045611014]: [cf1] SUP: Can not fly
[ INFO] [1703676634.053410814]: [cf1] SUP: Ready to fly
[ INFO] [1703676634.089081236]: [cf1] ESTKALMAN: State out of boundsESTKALMAN: State out of bounds, resetting
[ INFO] [1703676634.101465650]: [cf1] Requesting parameters...
[ INFO] [1703676634.531001242]: Found variables in cache.
[ INFO] [1703676637.612809220]: [cf1] reqParamTOC: 4.034623 s (279 params)
[ INFO] [1703676637.612860207]: [cf1] Requesting logging variables...
[ INFO] [1703676637.653057271]: Found variables in cache.
[ INFO] [1703676637.654392556]: [cf1] reqLogTOC: 0.041586 s
[ INFO] [1703676637.706149711]: [cf1] logBlocks: 4.034623 s
[ INFO] [1703676637.751125280]: Requesting memories...
[ INFO] [1703676637.803832672]: Memories: 9
[ INFO] [1703676638.046434887]: [cf1] Ready. Elapsed: 4.468248 s
[ INFO] [1703676638.046481122]: CF run: 4.468411 s
update commander/enHighLevel to 1
update kalman/resetEstimation to 1
update locSrv/extPosStdDev to 0.001
update locSrv/extQuatStdDev to 0.05
update ring/effect to 16
update ring/headlightEnable to 0
update ring/solidBlue to 255
update ring/solidGreen to 0
update ring/solidRed to 0
update stabilizer/controller to 2
update stabilizer/estimator to 2
update ctrlMel/i_range_m_z to 1500
update ctrlMel/i_range_xy to 2
update ctrlMel/i_range_z to 0.4
update ctrlMel/kR_xy to 70000
update ctrlMel/kR_z to 60000
update ctrlMel/kd_omega_rp to 200
update ctrlMel/kd_xy to 0.2
update ctrlMel/kd_z to 0.4
update ctrlMel/ki_m_z to 500
update ctrlMel/ki_xy to 0.05
update ctrlMel/ki_z to 0.05
update ctrlMel/kp_xy to 0.4
update ctrlMel/kp_z to 1.25
update ctrlMel/kw_xy to 20000
update ctrlMel/kw_z to 12000
update ctrlMel/mass to 0.032
update ctrlMel/massThrust to 132000
[ INFO] [1703676638.061793499]: [cf1] Update parameters
[ INFO] [1703676638.506149261]: [cf1] CONTROLLER: Using Mellinger (2) controller
[ INFO] [1703676639.462156270]: Update params: 1.415631 s
[ INFO] [1703676639.462372777]: Started 1 threads
terminate called after throwing an instance of 'std::runtime_error'
  what():  LIBUSB_ERROR_TIMEOUT
[crazyswarm_server-1] process has died [pid 7901, exit code -6, cmd /home/bitcraze/crazyswarm/ros_ws/devel/lib/crazyswarm/crazyswarm_server __name:=crazyswarm_server __log:=/home/bitcraze/.ros/log/b82dc00c-a495-11ee-b779-13d910275467/crazyswarm_server-1.log].
log file: /home/bitcraze/.ros/log/b82dc00c-a495-11ee-b779-13d910275467/crazyswarm_server-1*.log

Upon executing the command, RViz opens up, and the Crazyradio 2.0 starts to show a green light. In the RViz window, an airplane's axes are displayed with the label "world", and it seems to be preparing to launch a cf1 drone.

However, the process eventually fails with the error message: [crazyswarm_server-1] process has died [pid 7901, exit code -6, ...].

I am wondering if this issue could be related to an incorrect update of the crazyradio-firmware (I have asked a separate question regarding this in another discussion).

Thank you for taking the time to address my query!

**Comment by whoenig** [comment]
Crazyswarm1 has never been tested with Crazyradio 2.0. From the error message, it looks like you might need to upgrade your Crazyradio firmware.

**Reply by nfu64967512** [reply]
Hi, so I should use crazyradio PA instead? I asked about crazyradio-firmware update problem on bitcraze discussions, maybe I should wait for their answer? Because every time I try to update my crazyradio 2.0 by  "https://github.com/bitcraze/crazyradio-firmware" and always get failed.  Thank u for solving my problem!
BTW shouldn't I use LPS position for crazyswarm1? I seldom see there is anyone use this position way to swarm.

**Reply by whoenig** [reply, marked as answer]
Crazyradio2 should work with the latest firmware from https://github.com/bitcraze/crazyradio2-firmware, but hasn't been tested by me.
Most people use a motion capture, but LPS and Lighthouse should work too. LPS is the least accurate of these options.

**Reply by nfu64967512** [reply]
OK! I will try it later! I will try your way.
Thanks!

**Reply by jpreiss** [reply]
FYI for future readers: I've been using Crazyswarm1 with the Crazyradio 2.0 recently and have observed no problems.
